File: src/slmporting/utils/units.py
from enum import Enum


class Unit(Enum):
    METERS      = 1.0
    MILLIMETERS = 1e-3
    MICROMETERS = 1e-6
    NANOMETERS  = 1e-9


class Length:
    def __init__(self, mantissa: float, unit: Unit = Unit.METERS):
        self.value = mantissa * unit.value

    def convert_to(self, unit: Unit) -> float:
        return self.value / unit.value

File: cpp/src/units.cpp
#include "units.h"
#include <cmath>


Length::Length(const double mantissa, const Unit unit) : value(
	mantissa * std::pow(10.0, static_cast<int>(unit))) {}


double Length::convert_to(Unit unit) {
	return value * std::pow(10.0, -static_cast<int>(unit));
}
